lua/autorun/server/sv_restricted_propcore_functions.lua

```diff
@@ -0,0 +1,44 @@
+-- Restricted Propcore Functions
+---------------------------------------------------------------------------------------
+RESTRICTED_FUNCTIONS = {
+  'propSpawn(sn)',
+  'propSpawn(en)',
+  'propSpawn(svn)',
+  'propSpawn(evn)',
+  'propSpawn(san)',
+  'propSpawn(ean)',
+  'propSpawn(svan)',
+  'propSpawn(evan)',
+  'seatSpawn(sn)',
+  'seatSpawn(svan)',
+  'setPos(e:v)',
+  'propManipulate(e:vannn)',
+  'reposition(e:v)'
+}
+---------------------------------------------------------------------------------------
+
+-- Rank Restriction Tables
+---------------------------------------------------------------------------------------
+-- Admin rank restricted functions
+ADMIN_RESTRICTED_FUNCTIONS = {}
+
+-- Regular rank restricted functions
+REGULAR_RESTRICTED_FUNCTIONS = ADMIN_RESTRICTED_FUNCTIONS
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(sn)']           = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(en)']           = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(svn)']          = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(evn)']          = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(san)']          = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(ean)']          = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(svan)']         = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(evan)']         = {build = true, pvp = true}
```

Comment:
I have an idea for this.

Keep the full function signature (including parameters) in `RESTRICTED_FUNCTIONS`, but do something like this for the insertion into rank tables:
```lua
REGULAR_RESTRICTED_FUNCTIONS['propSpawn'] = {build = true, pvp = true}
```
And when checking the restrictions in the code, `gsub` the (parameters) from the function's signature, and check it then.
Unless you think there will be cases where we'll give someone (for example) `propSpawn(sn)` but not `propSpawn(en)`?